core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSail.java

```diff
@@ -926,4 +930,60 @@ public long getValidationResultsLimitTotal() {
 	public void setValidationResultsLimitTotal(long validationResultsLimitTotal) {
 		this.validationResultsLimitTotal = validationResultsLimitTotal;
 	}
+
+	@Override
+	public IsolationLevel getDefaultIsolationLevel() {
+		return super.getDefaultIsolationLevel();
+	}
+
+	@Override
+	public Optional<TransactionSetting> internTransactionSetting(String name, String value) {
+		if (Settings.ValidationApproach.Disabled.getName().equals(name)
+				&& Settings.ValidationApproach.Disabled.getValue().equals(value)) {
+			return Optional.of(Settings.ValidationApproach.Disabled);
+		}
+		if (Settings.ValidationApproach.Bulk.getName().equals(name)
+				&& Settings.ValidationApproach.Bulk.getValue().equals(value)) {
+			return Optional.of(Settings.ValidationApproach.Bulk);
+		}
+		return getBaseSail().internTransactionSetting(name, value);
+
+	}
```

Comment:
I've introduced this concept here for handling transaction settings that come from http. Essentially we "intern" them just like you can intern a string. This makes it much easier to write the rest of the code since we can use things like `instanceOf` and `==` for enums. 

Also this opens up for supporting multiple isolation levels. I was thinking of adding our current isolation levels to the abstract sail class, so we retrieve them from there if possible. That way another sail can override that class and provide their own isolation levels. What do you think? @jeenbroekstra

Reply:
I don't quite follow what you have in mind for multiple isolation levels. How about we park that for now? Let's get the basic transaction settings working first.

Reply:
As for the 'interning': it's a workable approach but I had an alternative with a registry for settings and a factory in mind, so that (de)serializing of TransactionSettings can happen in the server application code and is not a responsibility for the Sail itself. I'll try and sketch something.